langchain4j/src/main/java/dev/langchain4j/service/AiServices.java

```diff
@@ -250,14 +451,29 @@ private void addIfNeeded(ChatMessage systemMessage) { // TODO move to memory?
                             }
                         }
                         if (shouldAddSystemMessage) {
-                            context.chatMemory.add(systemMessage);
+                            chatMemory.add(systemMessage);
                         }
                     }
                 });
 
         return (T) proxyInstance;
     }
 
+    private Optional<Object> getUserId(Method method, Object[] args) {
```

Comment:
should we extract some methods into another util class?

Reply:
yes, but this will be another PR where I will refactor the whole AiServices class